url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/CameraAgent.java

```diff
@@ -0,0 +1,554 @@
+					 * NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN)) {
+					 */
+					if (nests.get(i).state == NestStatus.READY
+							|| nests.get(i).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+							|| nests.get(i).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+						if (nests.get(i + 1).state == NestStatus.READY
+								|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+								|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+							print("Taking photos of nests");
+							takePictureOfNest(nests.get(i), nests.get(i + 1));
+							return true;
+						} else if (nests.get(i + 1).state == NestStatus.NOT_READY) {
+							print("Started a timer for nest " + (i + 1));
+							nests.get(i + 1).state = NestStatus.STARTED_TIMER;
+							final Nest tempNest = nests.get(i + 1).nest;
+							timer.schedule(new TimerTask() {
+								@Override
+								public void run() {
+									msgIAmFull(tempNest);
+								}
+							}, 1000);
+							return true;
+						}
+					} else if (nests.get(i + 1).state == NestStatus.READY
+							|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH
+							|| nests.get(i + 1).state == NestStatus.READY_TO_RE_PHOTOGRAPH_AGAIN) {
+						if (nests.get(i).state == NestStatus.NOT_READY) {
+							print("Started a timer for nest " + i);
+							nests.get(i).state = NestStatus.STARTED_TIMER;
+							final Nest tempNest = nests.get(i).nest;
+							timer.schedule(new TimerTask() {
+								@Override
+								public void run() {
+									msgIAmFull(tempNest);
+								}
+							}, 1000);
+							return true;
+						}
+					}
+					// }
+				}
+			}
+		}
+
+		synchronized (nests) {
+			for (MyNest n : nests) {
+				// print("nest "+n.nest.currentPartType+ " state is "+n.state);
+				if (n.state == NestStatus.PHOTOGRAPHED || n.state == NestStatus.RE_PHOTOGRAPHED_ONCE
+						|| n.state == NestStatus.RE_PHOTOGRAPHED_TWICE) {
+					tellPartsRobot(n);
+					return true;
+				}
+			}
+		}
+
+		if (mk != null) {
+			if (mk.ks == KitStatus.NOT_READY) {
+				takePictureOfKit(mk);
+				return true;
+			} else if (mk.ks == KitStatus.DONE) {
+				tellKitRobot(mk);
+				return true;
+			} else if (mk.ks == KitStatus.FAILED) {
+				tellKitRobotFailure(mk);
+				return true;
+			}
+
+		}
+
+		return false;
+	}
+
+	/*
+	 * Actions
+	 */
+	private void tellKitRobot(MyKit kit) {
+		mk = null;
+		kitRobot.msgKitPassedInspection();
+		stateChanged();
+	}
+
+	private void tellKitRobotFailure(MyKit kit) {
+		mk = null;
+		kitRobot.msgKitFailedInspection();
+		stateChanged();
+	}
+
+	private void takePictureOfKit(MyKit kit) {
+		mk.ks = KitStatus.PICTURE_BEING_TAKEN;
+		if (guiCamera != null) {
+			guiCamera.takeKitPhoto(kit.kit.kitGraphics);
+			print("Blocking");
+			try {
+				animation.acquire();
+			} catch (InterruptedException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			print("Got permit");
+		}
+		stateChanged();
+	}
+
+	private void tellPartsRobot(final MyNest n) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.